src/backend/utils/adt/datetime.c

```diff
@@ -1882,6 +1882,7 @@ DecodeTimeOnly(char **field, int *ftype, int nf,
 						case DTK_DAY:
 							if (tzp == NULL)
 								return DTERR_BAD_FORMAT;
+							FALL_THROUGH
```

Comment:
I suggest to remove ```default: break;``` and don't add `FALL_THROUGH`

Reply:
Done